src/rules/value-keyword-case/index.js

```diff
@@ -0,0 +1,67 @@
+import valueParser from "postcss-value-parser"
+import {
+  cssWordIsVariable,
+  declarationValueIndexOffset,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "value-keyword-case"
+
+export const messages = ruleMessages(ruleName, {
+  expected: (actual, expected) => `Expected "${actual}" to be "${expected}"`,
+})
+
+const ignoredCharacters = new Set([
+  "+", "-", "/", "*", "%",
+])
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "lower",
+        "upper",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { value } = decl
+
+      valueParser(value).walk((node) => {
+        // Ignore keywords within `url` function
+        if (node.type === "function" && node.value === "url") { return false }
+
+        const keyword = node.value
+        const hexMatch = /^#[0-9A-Za-z]+/.exec(keyword)
+
+        // Ignore css variables, and hex values, and math operators, and sass interpolation
+        if (node.type !== "word"
+          || cssWordIsVariable(node.value)
+          || hexMatch
+          || ignoredCharacters.has(keyword)
```

Comment:
@davidtheclark performance. `valueParser.unit` spends time on parse. i think we not should spends this time on arithmetic operators (scss, less).